python/pyspark/worker.py

```diff
@@ -177,8 +180,11 @@ def process():
             process()
     except Exception:
         try:
+            exc_info = traceback.format_exc()
+            if isinstance(exc_info, unicode):
+                exc_info = exc_info.encode('utf-8')
```

Comment:
I guess we need to follow #17267 each other to fix correctly.

Reply:
@HyukjinKwon this pr can only be hanging?

Reply:
cc @jiangxb1987